### ISSUE TITLE
[Package Request] - Squid

### ISSUE THREAD

**Issue by rerooted** [amazonlinux/amazon-linux-2023#152]
**What package is missing from Amazon Linux 2022? Please describe and include package name.**

Squid

**Is this an update to existing package or new package request?**

New package

**Is this package available in Amazon Linux 2?**

Yes

**Number of users impacted**

Unknown

Squid is useful to provide HTTP proxy support for CodeBuild, non-public IP enabled hosts, etc.  CodeBuild does not support a public IP [1] and without a NAT gateway [2] it is quite problematic to reach the Internet to download items.  Having a proxy works around these issues and saves the cost of having a NAT gateway in a VPC.

[1] https://docs.aws.amazon.com/codebuild/latest/userguide/use-proxy-server.html
[2] https://docs.aws.amazon.com/codebuild/latest/userguide/vpc-support.html#best-practices-for-vpcs

**Comment by devinmitchell** [comment]
Squid v5+ would be a most welcomed addition.

**Comment by ozbenh** [comment]
Are there specific bugs you are concerned about ?  6+ isn't LTS either and other relevant distros also seem to be staying on 5.x at this point, so there are sources of bug fixes available.

**Comment by Nevon** [comment]
[The 5.X line support ended a month ago](https://endoflife.date/squid). Perhaps individual distros are back porting patches, but not the squid developers themselves. I found out as I reported bug [5294](https://bugs.squid-cache.org/show_bug.cgi?id=5294) originally for the version of squid that's shipping in AL2023, and was told the 5 release isn't getting updates anymore and to verify if the bug is also present in 6. There is an incoming patch which will only land in 6, unless back ported by distros.
